wizeline/falcon/middlewares/bodyParser.py

```diff
@@ -0,0 +1,58 @@
+import json
+from json.decoder import JSONDecodeError
+
+import falcon
+
+class BodyParserMiddleware:
+    def process_resource(self, req, resp, resource, params):
+        if self._is_middleware_enabled(resource) and self._request_method_has_payload(req):
+            if self._is_json_content_type(req):
+                try:
+                    req.text = self._get_payload(req)
+
+                    if req.text.strip() != '':
+                        req.json = json.loads(req.text)
+                    else:
+                        req.json = {}
+
+                except JSONDecodeError:
+                    raise falcon.HTTPInternalServerError()
+            elif self._is_urlencoded_content_type(req):
+                req.json = req.params
+            else:
+                raise falcon.HTTPUnsupportedMediaType()
+
+    def _is_middleware_enabled(self, resource):
+        return not hasattr(resource, 'disable_json_middleware') \
+            or not resource.disable_json_middleware
+
+    def _request_method_has_payload(self, req):
+        return req.method in ('POST', 'PUT', 'PATCH')
```

Comment:
We can have `POST` `PUT` and `PATCH` methods without payload, so I'd prefer to have another name for this function to avoid confusion